test/RpcCompatibility.js

```diff
@@ -484,4 +484,21 @@ describe('RpcCompatibility', () => {
             await checkReceipt(receipt, newValue);
         });
     });
+
+    it('should throw error when querying eth_getLogs with a future blockHash', async () => {
+        const currentBlockNumber = await provider.getBlockNumber();
+        const futureBlockNumber = currentBlockNumber + 1000;
+
+        const fakeFutureBlockHash = ethers.utils.hexZeroPad(ethers.utils.hexlify(futureBlockNumber), 32);
```

Comment:
There's no way to predict a future block hash so this is effectively checking for an unknown block hash and not a future one? Also, can random 32 bytes suffice instead of trying to encode a future block number?